boot/assets.go

```diff
@@ -568,10 +568,10 @@ func (o *TrustedAssetsUpdateObserver) observeRollback(bl bootloader.Bootloader,
 	}
 
 	if err := o.modeenv.Write(); err != nil {
-		return false, fmt.Errorf("cannot write modeeenv: %v", err)
+		return gadget.ChangeAbort, fmt.Errorf("cannot write modeeenv: %v", err)
 	}
 
-	return false, nil
```

Comment:
the intention expressed by false here (it was anyway not yet used) was wrong?  will we even use the value of ContentChangeAction for rollback?

Reply:
The use of `false` here was wrong, I tried to have the code return `true` for al the cases where it's fine to proceed with the change.

As for the rollback scenario, I don't see any need for inspecting the returned value at this point.